python/pyspark/sql/tests.py

```diff
@@ -832,6 +832,21 @@ def filename(path):
         row2 = df2.select(sameText(df2['file'])).first()
         self.assertTrue(row2[0].find("people.json") != -1)
 
+    def test_input_file_name_reset_for_rdd(self):
+        rdd = self.sc.textFile('python/test_support/hello/hello.txt').map(lambda x: {'data': x})
+        df = self.spark.createDataFrame(rdd, "data STRING")
+        df.select(input_file_name().alias('file')).collect()
```

Comment:
@jose-torres, seems `input_file_name` is not defined in this file

Reply:
Hmm, I don't know why the import didn't get copied over. Added.